src/shogun/kernel/string/HistogramWordStringKernel.cpp

```diff
@@ -434,23 +434,17 @@ void HistogramWordStringKernel::init()
 	/*m_parameters->add_vector(&variance, &num_params2, "variance");*/
 	watch_param("variance", &variance, &num_params2);
 
-	SG_ADD(&estimate, "estimate", "Plugin Estimate.");
-
-	add_callback_function("estimate", [this](){
-		if (!std::dynamic_pointer_cast<PluginEstimate>(estimate)) {
-			error("Expected Machine to be PluginEstimate");
-			estimate.reset();
-		}
-	});
+	SG_ADD(&estimate, "estimate", "Plugin Estimate.", ParameterProperties::CONSTRAIN,
+		SG_CONSTRAINT(dynamic_cast_checker<std::shared_ptr<Machine>, PluginEstimate>()));
```

Comment:
semantics are here that anything placed with put should be a subclass of Machine, but the type stored is PluginEstimate? Then the latter would be kind of redundant in the specification... is it necessary? Because put already checks that?

Reply:
Hmmm I guess I can accept SGObject type